deploy/docker/Dockerfile.api

```
@@ -0,0 +1,53 @@
+# Copy the binary
+COPY --from=builder /app/main /main
+
+# Create non-root user (using numeric ID for scratch image)
+USER 65534:65534
+
+# Expose port
+EXPOSE 8080
+
+# Health check - using the --health flag that exists in your main.go
+HEALTHCHECK --interval=30s --timeout=10s --start-period=5s --retries=3 \
+    CMD ["/main", "--health"] || exit 1
```

Hadolint (2.12.0)

[error] unexpected '|'
expecting '\', a new line followed by the next instruction, or at least one space

(DL1000)